File: data/src/notFoss/java/org/cryptomator/data/cloud/googledrive/GoogleDriveFolder.java
package org.cryptomator.data.cloud.googledrive;

import org.cryptomator.domain.Cloud;
import org.cryptomator.domain.CloudFolder;

class GoogleDriveFolder implements CloudFolder, GoogleDriveNode {

	private final GoogleDriveFolder parent;
	private final String name;
	private final String path;
	private final String driveId;

	public GoogleDriveFolder(GoogleDriveFolder parent, String name, String path, String driveId) {
		this.parent = parent;
		this.name = name;
		this.path = path;
		this.driveId = driveId;
	}

	@Override
	public Cloud getCloud() {
		return parent.getCloud();
	}

	@Override
	public String getName() {
		return name;
	}

	@Override
	public String getPath() {
		return path;
	}

	@Override
	public String getDriveId() {
		return driveId;
	}

	@Override
	public GoogleDriveFolder getParent() {
		return parent;
	}

	@Override
	public GoogleDriveFolder withCloud(Cloud cloud) {
		return new GoogleDriveFolder(parent.withCloud(cloud), name, path, driveId);
	}
}

File: data/src/apkStorePlaystore/java/org/cryptomator/data/cloud/googledrive/GoogleDriveFolder.kt
package org.cryptomator.data.cloud.googledrive

import org.cryptomator.domain.Cloud
import org.cryptomator.domain.CloudFolder

open class GoogleDriveFolder(override val parent: GoogleDriveFolder?, override val name: String, override val path: String, override val driveId: String?) : CloudFolder, GoogleDriveNode {

	override val cloud: Cloud?
		get() = parent?.cloud

	override fun withCloud(cloud: Cloud?): GoogleDriveFolder? {
		return GoogleDriveFolder(parent?.withCloud(cloud), name, path, driveId)
	}
}
